pgoapi/pgoapi.py

```diff
@@ -408,37 +417,52 @@ def get_caught_pokemons(self, inventory_items):
                     caught_pokemon[pokemon.pokemon_id].append(pokemon)
         return caught_pokemon
 
+    def do_release_pokemon(self, pokemon):
+        self.log.info("Releasing pokemon: %s", pokemon)
+        self.release_pokemon(pokemon_id=pokemon.id)
+        release_res = self.call()['responses']['RELEASE_POKEMON']
+        status = release_res.get('result', -1)
+        if status == 1:
+            self.log.info("Successfully Released Pokemon %s", pokemon)
+        else:
+            self.log.debug("Failed to release pokemon %s, %s", pokemon, release_res)
+            self.log.info("Failed to release Pokemon %s", pokemon)
+        sleep(1)
+
     def cleanup_pokemon(self, inventory_items=None):
         if not inventory_items:
                 inventory_items = self.get_inventory().call()['responses']['GET_INVENTORY']['inventory_delta'][
                     'inventory_items']
         caught_pokemon = self.get_caught_pokemons(inventory_items)
 
         for pokemons in caught_pokemon.values():
-            # Only if we have more than MIN_SIMILAR_POKEMON
             if len(pokemons) > self.MIN_SIMILAR_POKEMON:
-                pokemons = sorted(pokemons, key=lambda x: (x.cp, x.iv), reverse=True)
-                # keep the first pokemon....
-                for pokemon in pokemons[self.MIN_SIMILAR_POKEMON:]:
-                    if self.is_pokemon_eligible_for_transfer(pokemon):
-                        self.log.info("Releasing pokemon: %s", pokemon)
-                        self.release_pokemon(pokemon_id=pokemon.id)
-                        release_res = self.call()['responses']['RELEASE_POKEMON']
-                        status = release_res.get('result', -1)
-                        if status == 1:
-                            self.log.info("Successfully Released Pokemon %s", pokemon)
-                        else:
-                            self.log.debug("Failed to release pokemon %s, %s", pokemon, release_res)
-                            self.log.info("Failed to release Pokemon %s", pokemon)
-                        sleep(3)
-
-    def is_pokemon_eligible_for_transfer(self, pokemon):
-        return (pokemon.pokemon_id in self.throw_pokemon_ids and not pokemon.is_favorite) \
-               or (not pokemon.is_favorite and
-                   pokemon.iv < self.MIN_KEEP_IV and
-                   pokemon.cp < self.KEEP_CP_OVER and
-                   pokemon.is_valid_pokemon() and
-                   pokemon.pokemon_id not in self.keep_pokemon_ids)
+                # highest CP pokemon first
+                pokemons = sorted(pokemons, key=lambda pok: (pok.cp, pok.iv), reverse=True)
+                pokemons_keep_counter = 0
+                pokemons_keep_iv_counter = 0
+
+                for pokemon in pokemons:
+                    # never release favorites and other defined pokemons
+                    if pokemon.is_favorite or pokemon.pokemon_id in self.keep_pokemon_ids:
+                        pokemons_keep_counter += 1
```

Comment:
can we put the logic to check for eligibiluty in the method is_pokemon_eligible_for_transfer to keep it cleaner and modular .